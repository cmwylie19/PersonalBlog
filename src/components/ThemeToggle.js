import React from "react"
import Toggle from "react-toggle"

const ThemeToggle = props => (
  <div style={{ display: "flex", justifyContent: "center" }}>
    {props.theme ? "Light  " : "Dark "}
    <div style={{ width: "10px" }}></div>{" "}
    <Toggle
      defaultChecked={props.theme}
      aria-label="Toggle Theme"
      onChange={props.toggleTheme}
    />
  </div>
)
export default ThemeToggle
