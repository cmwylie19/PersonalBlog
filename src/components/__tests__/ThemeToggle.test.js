import React from "react"
import renderer from "react-test-renderer"

import ThemeToggle from "../ThemeToggle"

describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<ThemeToggle theme={true} toggleTheme={() => true} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})