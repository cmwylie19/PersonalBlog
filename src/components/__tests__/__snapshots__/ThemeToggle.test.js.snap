// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "justifyContent": "center",
    }
  }
>
  Light  
  <div
    style={
      Object {
        "width": "10px",
      }
    }
  />
   
  <div
    className="react-toggle react-toggle--checked"
    onClick={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="react-toggle-track"
    >
      <div
        className="react-toggle-track-check"
      >
        <svg
          height="11"
          viewBox="0 0 14 11"
          width="14"
        >
          <title>
            switch-check
          </title>
          <path
            d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
            fill="#fff"
            fillRule="evenodd"
          />
        </svg>
      </div>
      <div
        className="react-toggle-track-x"
      >
        <svg
          height="10"
          viewBox="0 0 10 10"
          width="10"
        >
          <title>
            switch-x
          </title>
          <path
            d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
            fill="#fff"
            fillRule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      className="react-toggle-thumb"
    />
    <input
      aria-label="Toggle Theme"
      className="react-toggle-screenreader-only"
      defaultChecked={true}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="checkbox"
    />
  </div>
</div>
`;
